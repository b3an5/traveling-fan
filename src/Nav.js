import React, { Component } from 'react';
import './Nav.scss';

export default class Nav extends Component {
  constructor() {
    super();
    this.state = {
      i: null
    }
  }

  render() {
    return (
      <nav className="nav-area">
        <form action="#">
          <input type="text" placeholder="Enter team name or city" />
          <label for="division-select">Select an NFL Division:</label>

          <select id="division-select">
            <option value="afc">--AFC Division--</option>
            <option value="afc-north">AFC North</option>
            <option value="afc-east">AFC East</option>
            <option value="afc-south">AFC South</option>
            <option value="afc-west">AFC West</option>
            <option value="nfc">--NFC Division--</option>
            <option value="nfc-north">NFC North</option>
            <option value="nfc-east">NFC East</option>
            <option value="nfc-south">NFC South</option>
            <option value="nfc-west">NFC West</option>
          </select>

        </form>
      </nav>
    )
  }
}