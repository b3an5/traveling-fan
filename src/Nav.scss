nav {
  background: url('./images/nav-bg.png');
  background-size: cover;
  height: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
}

form input, form label {
  margin: 0 80px;
}

form label {
  font-size: 2rem;

}

input {
  height: 50px;
  width: 300px;
  font-size: 1.3rem;
}

select {
  font-size: 1.3rem;
  height: 50px;
  width: 200px;
}