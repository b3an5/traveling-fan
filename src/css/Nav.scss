nav {
  height: 120px;
  display: flex;
  justify-content: center;
  align-items: center;
}

form {
  display: flex;
  width: 100%;
  justify-content: space-evenly;
}

form label {
  font-size: 3rem;
  font-weight: 900;
  padding-top: 8px;
  margin: 0 -15%;
  color: #DA2227;
  text-shadow:
    -1px -1px 0 #fff,
    1px -1px 0 #fff,
    -1px 1px 0 #fff,
    1px 1px 0 #fff;
}

input {
  height: 50px;
  width: 300px;
  font-size: 2rem;
  font-weight: 900;
  color: #003B73;
  text-shadow:
		-1px -1px 0 #fff,
		1px -1px 0 #fff,
		-1px 1px 0 #fff,
		1px 1px 0 #fff;
  background-color: white;
  padding-left: 20px;
  &:hover {
    background-color: #DA2227;
    border: 5px solid white;
  }
  &::placeholder {
    color: #003B73;
  }
  border: 5px solid #DA2227;
  border-radius: 15px;
}

select{
  font-size: 1.5rem;
  font-weight: 900;
  width: 331px;
  height: 62px;
  padding: 5px;
  color: #003B73;
  text-shadow:
  -1px -1px 0 #fff,
  1px -1px 0 #fff,
  -1px 1px 0 #fff,
  1px 1px 0 #fff;
  border: 5px solid #DA2227;
  border-radius: 15px;
  background-color: white;
  &:hover {
    background-color: #DA2227;
    border: 5px solid white;
  }
}

select option { 
  color: black;
  &:first-child {
    background-color: #DA2227;
  }
}