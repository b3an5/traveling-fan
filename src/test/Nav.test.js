import React from 'react';
import Nav from '../Nav.js';
import { shallow } from 'enzyme';


const fakeNflTeam = {
    "name": "Arizona Cardinals",
    "state": "Arizona",
    "city": "Glendale",
    "team_icon": "http://assets.stickpng.com/thumbs/580b585b2edbce24c47b2aff.png",
    "division": "NFC West",
    "team_img": "https://cdn.vox-cdn.com/thumbor/M5Gx-DCa9eRAqw_aqIyuG6I2wDo=/0x0:4777x3021/1200x800/filters:focal(2094x550:2858x1314)/cdn.vox-cdn.com/uploads/chorus_image/image/62274919/usa_today_11648254.0.jpg"
  }

const mockFindSearchedTeam = jest.fn()
const mockFindSelectedDivision = jest.fn()
describe('Nav', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Nav findSearchedTeam={mockFindSearchedTeam}
           findSelectedDivision={mockFindSelectedDivision} />
      )
  })

  it('should match the snapshot with all data passed correctly to the Nav', () => {
      expect(wrapper).toMatchSnapshot();
  })

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual({team: null, selectedValue: 'init'})
  })

  it('should set the state with a selected divison/conference from the dropdown menu', () => {
    expect(wrapper.state()).toEqual({team: null, selectedValue: 'init'})
    wrapper.find('select').simulate('change', {target: {value : 'NFC'}})
    expect(wrapper.find('select').props().value).toBe('NFC')
    expect(wrapper.state()).toEqual({team: null, selectedValue: 'NFC'})
  })

  it('should set the state with a selected team when a team is entered into the search input field', () => {
    expect(wrapper.state()).toEqual({team: null, selectedValue: 'init'})
    wrapper.find('input').simulate('change', {target: {value : 'Arizona'}})
    expect(wrapper.state()).toEqual({team: "Arizona", selectedValue: 'init'})
  })
}) 
