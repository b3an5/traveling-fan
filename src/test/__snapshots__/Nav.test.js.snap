// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot with all data passed correctly to the Nav 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    findSearchedTeam={[MockFunction]}
    findSelectedDivision={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        action="#"
      >
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Enter team name"
          type="text"
        />
        <label
          htmlFor="division-select"
        >
          -- OR --
        </label>
        <select
          id="division-select"
          onChange={[Function]}
          value="init"
        >
          <option
            className="hello"
            disabled={true}
            value="init"
          >
            Select a Division: 
          </option>
          <option
            value="all"
          >
            Show All
          </option>
          <option
            disabled={true}
          />
          <option
            value="AFC"
          >
            AFC Conference
          </option>
          <option
            value="AFC North"
          >
            AFC North
          </option>
          <option
            value="AFC East"
          >
            AFC East
          </option>
          <option
            value="AFC South"
          >
            AFC South
          </option>
          <option
            value="AFC West"
          >
            AFC West
          </option>
          <option
            disabled={true}
          />
          <option
            value="NFC"
          >
            NFC Conference
          </option>
          <option
            value="NFC North"
          >
            NFC North
          </option>
          <option
            value="NFC East"
          >
            NFC East
          </option>
          <option
            value="NFC South"
          >
            NFC South
          </option>
          <option
            value="NFC West"
          >
            NFC West
          </option>
        </select>
      </form>,
      "className": "nav-area",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "#",
        "children": Array [
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Enter team name"
            type="text"
          />,
          <label
            htmlFor="division-select"
          >
            -- OR --
          </label>,
          <select
            id="division-select"
            onChange={[Function]}
            value="init"
          >
            <option
              className="hello"
              disabled={true}
              value="init"
            >
              Select a Division: 
            </option>
            <option
              value="all"
            >
              Show All
            </option>
            <option
              disabled={true}
            />
            <option
              value="AFC"
            >
              AFC Conference
            </option>
            <option
              value="AFC North"
            >
              AFC North
            </option>
            <option
              value="AFC East"
            >
              AFC East
            </option>
            <option
              value="AFC South"
            >
              AFC South
            </option>
            <option
              value="AFC West"
            >
              AFC West
            </option>
            <option
              disabled={true}
            />
            <option
              value="NFC"
            >
              NFC Conference
            </option>
            <option
              value="NFC North"
            >
              NFC North
            </option>
            <option
              value="NFC East"
            >
              NFC East
            </option>
            <option
              value="NFC South"
            >
              NFC South
            </option>
            <option
              value="NFC West"
            >
              NFC West
            </option>
          </select>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Enter team name",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-- OR --",
            "htmlFor": "division-select",
          },
          "ref": null,
          "rendered": "-- OR --",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                className="hello"
                disabled={true}
                value="init"
              >
                Select a Division: 
              </option>,
              <option
                value="all"
              >
                Show All
              </option>,
              <option
                disabled={true}
              />,
              <option
                value="AFC"
              >
                AFC Conference
              </option>,
              <option
                value="AFC North"
              >
                AFC North
              </option>,
              <option
                value="AFC East"
              >
                AFC East
              </option>,
              <option
                value="AFC South"
              >
                AFC South
              </option>,
              <option
                value="AFC West"
              >
                AFC West
              </option>,
              <option
                disabled={true}
              />,
              <option
                value="NFC"
              >
                NFC Conference
              </option>,
              <option
                value="NFC North"
              >
                NFC North
              </option>,
              <option
                value="NFC East"
              >
                NFC East
              </option>,
              <option
                value="NFC South"
              >
                NFC South
              </option>,
              <option
                value="NFC West"
              >
                NFC West
              </option>,
            ],
            "id": "division-select",
            "onChange": [Function],
            "value": "init",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select a Division: ",
                "className": "hello",
                "disabled": true,
                "value": "init",
              },
              "ref": null,
              "rendered": "Select a Division: ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Show All",
                "value": "all",
              },
              "ref": null,
              "rendered": "Show All",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "AFC Conference",
                "value": "AFC",
              },
              "ref": null,
              "rendered": "AFC Conference",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "AFC North",
                "value": "AFC North",
              },
              "ref": null,
              "rendered": "AFC North",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "AFC East",
                "value": "AFC East",
              },
              "ref": null,
              "rendered": "AFC East",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "AFC South",
                "value": "AFC South",
              },
              "ref": null,
              "rendered": "AFC South",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "AFC West",
                "value": "AFC West",
              },
              "ref": null,
              "rendered": "AFC West",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NFC Conference",
                "value": "NFC",
              },
              "ref": null,
              "rendered": "NFC Conference",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NFC North",
                "value": "NFC North",
              },
              "ref": null,
              "rendered": "NFC North",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NFC East",
                "value": "NFC East",
              },
              "ref": null,
              "rendered": "NFC East",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NFC South",
                "value": "NFC South",
              },
              "ref": null,
              "rendered": "NFC South",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NFC West",
                "value": "NFC West",
              },
              "ref": null,
              "rendered": "NFC West",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "form",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          action="#"
        >
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Enter team name"
            type="text"
          />
          <label
            htmlFor="division-select"
          >
            -- OR --
          </label>
          <select
            id="division-select"
            onChange={[Function]}
            value="init"
          >
            <option
              className="hello"
              disabled={true}
              value="init"
            >
              Select a Division: 
            </option>
            <option
              value="all"
            >
              Show All
            </option>
            <option
              disabled={true}
            />
            <option
              value="AFC"
            >
              AFC Conference
            </option>
            <option
              value="AFC North"
            >
              AFC North
            </option>
            <option
              value="AFC East"
            >
              AFC East
            </option>
            <option
              value="AFC South"
            >
              AFC South
            </option>
            <option
              value="AFC West"
            >
              AFC West
            </option>
            <option
              disabled={true}
            />
            <option
              value="NFC"
            >
              NFC Conference
            </option>
            <option
              value="NFC North"
            >
              NFC North
            </option>
            <option
              value="NFC East"
            >
              NFC East
            </option>
            <option
              value="NFC South"
            >
              NFC South
            </option>
            <option
              value="NFC West"
            >
              NFC West
            </option>
          </select>
        </form>,
        "className": "nav-area",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "#",
          "children": Array [
            <input
              className="search-input"
              onChange={[Function]}
              placeholder="Enter team name"
              type="text"
            />,
            <label
              htmlFor="division-select"
            >
              -- OR --
            </label>,
            <select
              id="division-select"
              onChange={[Function]}
              value="init"
            >
              <option
                className="hello"
                disabled={true}
                value="init"
              >
                Select a Division: 
              </option>
              <option
                value="all"
              >
                Show All
              </option>
              <option
                disabled={true}
              />
              <option
                value="AFC"
              >
                AFC Conference
              </option>
              <option
                value="AFC North"
              >
                AFC North
              </option>
              <option
                value="AFC East"
              >
                AFC East
              </option>
              <option
                value="AFC South"
              >
                AFC South
              </option>
              <option
                value="AFC West"
              >
                AFC West
              </option>
              <option
                disabled={true}
              />
              <option
                value="NFC"
              >
                NFC Conference
              </option>
              <option
                value="NFC North"
              >
                NFC North
              </option>
              <option
                value="NFC East"
              >
                NFC East
              </option>
              <option
                value="NFC South"
              >
                NFC South
              </option>
              <option
                value="NFC West"
              >
                NFC West
              </option>
            </select>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-input",
              "onChange": [Function],
              "placeholder": "Enter team name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "-- OR --",
              "htmlFor": "division-select",
            },
            "ref": null,
            "rendered": "-- OR --",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  className="hello"
                  disabled={true}
                  value="init"
                >
                  Select a Division: 
                </option>,
                <option
                  value="all"
                >
                  Show All
                </option>,
                <option
                  disabled={true}
                />,
                <option
                  value="AFC"
                >
                  AFC Conference
                </option>,
                <option
                  value="AFC North"
                >
                  AFC North
                </option>,
                <option
                  value="AFC East"
                >
                  AFC East
                </option>,
                <option
                  value="AFC South"
                >
                  AFC South
                </option>,
                <option
                  value="AFC West"
                >
                  AFC West
                </option>,
                <option
                  disabled={true}
                />,
                <option
                  value="NFC"
                >
                  NFC Conference
                </option>,
                <option
                  value="NFC North"
                >
                  NFC North
                </option>,
                <option
                  value="NFC East"
                >
                  NFC East
                </option>,
                <option
                  value="NFC South"
                >
                  NFC South
                </option>,
                <option
                  value="NFC West"
                >
                  NFC West
                </option>,
              ],
              "id": "division-select",
              "onChange": [Function],
              "value": "init",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Select a Division: ",
                  "className": "hello",
                  "disabled": true,
                  "value": "init",
                },
                "ref": null,
                "rendered": "Select a Division: ",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Show All",
                  "value": "all",
                },
                "ref": null,
                "rendered": "Show All",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                },
                "ref": null,
                "rendered": null,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "AFC Conference",
                  "value": "AFC",
                },
                "ref": null,
                "rendered": "AFC Conference",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "AFC North",
                  "value": "AFC North",
                },
                "ref": null,
                "rendered": "AFC North",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "AFC East",
                  "value": "AFC East",
                },
                "ref": null,
                "rendered": "AFC East",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "AFC South",
                  "value": "AFC South",
                },
                "ref": null,
                "rendered": "AFC South",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "AFC West",
                  "value": "AFC West",
                },
                "ref": null,
                "rendered": "AFC West",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                },
                "ref": null,
                "rendered": null,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NFC Conference",
                  "value": "NFC",
                },
                "ref": null,
                "rendered": "NFC Conference",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NFC North",
                  "value": "NFC North",
                },
                "ref": null,
                "rendered": "NFC North",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NFC East",
                  "value": "NFC East",
                },
                "ref": null,
                "rendered": "NFC East",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NFC South",
                  "value": "NFC South",
                },
                "ref": null,
                "rendered": "NFC South",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NFC West",
                  "value": "NFC West",
                },
                "ref": null,
                "rendered": "NFC West",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "form",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
